x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -416,7 +420,10 @@ const createThresholdTimeline = async (
 
     const alertDoc = formattedAlertData[0];
     const params = getField(alertDoc, ALERT_RULE_PARAMETERS);
-    const filters: Filter[] = params.filters ?? alertDoc.signal?.rule?.filters;
+    const filters: Filter[] =
+      (params as MightHaveFilters).filters ??
+      (alertDoc.signal?.rule as MightHaveFilters)?.filters ??
+      [];
```

Comment:
As @kqualters-elastic pointed out, the better fix might be to change the `filters` type in https://github.com/elastic/kibana/blob/7e7ff8239ce43c9a033b732a7adf4267e4a6c1f8/x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/types.ts#L163 from `filters?: Filter[]` to `filters: Filter[]`. Instead I chose to go with a more defensive approach in this PR in order to get this out with `8.3` because I am not familiar with the implication of the type changes to the detection engine.

If the reviewers feel strongly about actually fixing this in the detection engine, I propose to create a follow-up issue for those changes and assign the team that owns detection engine.

Let me know what you think.

Reply:
Ya I did this yesterday locally when seeing what was causing the problem, and I am cool with using either approach. Whichever we do should share types though, this is just asking for similar bugs to crop up over time.

Reply:
@kqualters-elastic I think fixing the type in the detection engine does also not fix the issue in this case since both reads (`getField(alertDoc, ALERT_RULE_PARAMETERS)` and `alertDoc.signal?.rule?.filters` return unsafe types (`any` and `unknown` respectively). We would have to fix the return type of `getField` I think.

Reply:
I don't think `interface MightHaveFilters` is not the most elegant solution but I can't come up with a better one just now :D